aea/skills/error/handler.py

```diff
@@ -63,10 +63,11 @@ def send_unsupported_protocol(self, envelope: Envelope) -> None:
         """
         logger.warning("Unsupported protocol: {}".format(envelope.protocol_id))
         reply = DefaultMessage(type=DefaultMessage.Type.ERROR,
-                               error_code=DefaultMessage.ErrorCode.UNSUPPORTED_PROTOCOL,
+                               error_code=DefaultMessage.ErrorCode.UNSUPPORTED_PROTOCOL.value,
                                error_msg="Unsupported protocol.",
                                error_data={"protocol_id": envelope.protocol_id})
-        self.context.outbox.put_message(to=envelope.sender, sender=self.context.agent_name, protocol_id=DefaultMessage.protocol_id,
+        self.context.outbox.put_message(to=envelope.sender, sender=self.context.agent_public_key,
+                                        protocol_id="error",
```

Comment:
I am not sure why this protocol_id has been replaced. @marcofavorito I think it should stay. Thoughts?

Reply:
Yes, it should stay.